src/core/server/elasticsearch/client/cluster_client.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import { getClientFacade } from './client_wrapper';
+import { ClientFacade } from './client_facade';
+import { configureClient } from './configure_client';
+import { Logger } from '../../logging';
+import { GetAuthHeaders, isRealRequest } from '../../http';
+import { Headers } from '../../http/router';
+import { ElasticsearchClientConfig } from './client_config';
+import { ScopedClusterClient, IScopedClusterClient } from './scoped_cluster_client';
+import { ScopeableRequest } from './types';
+import { ensureRawRequest, filterHeaders } from '../../http/router';
+
+const noop = () => undefined;
+
+interface IClusterClient {
+  asInternalUser: () => ClientFacade;
+  asScoped: (request: ScopeableRequest) => IScopedClusterClient;
+}
+
+export class ClusterClient implements IClusterClient {
+  private readonly internalWrapper: ClientFacade;
+  private readonly scopedClient: Client;
+
+  constructor(
+    private readonly config: ElasticsearchClientConfig,
+    logger: Logger,
+    private readonly getAuthHeaders: GetAuthHeaders = noop
+  ) {
+    this.internalWrapper = getClientFacade(configureClient(config, { logger }));
+    this.scopedClient = configureClient(config, { logger, scoped: true });
```

Comment:
let's stick to a common naming: wrapper or client

Reply:
Yes, this is a little unclear, but these are not the same type (related to https://github.com/elastic/kibana/pull/69905#discussion_r445523588)
- `internalWrapper` is the actual `ClientWrapper` instance
- `scopedClient` is the unique scoped client that will be used for every `scopedWrapper` in `asScoped`

Not sure I did the best approach addressing the 'avoid to parse the config multiple times' though, so maybe there is an alternative to that?

Reply:
Can we emphasise it in code ? :
```js
const internalClient = configureClient(config, { logger });
this.internalWrapper = getClientFacade(internalClient);
this.scopedClient = configureClient(config, { logger, scoped: true });
```